command/connect/envoy/envoy_test.go

```diff
@@ -560,15 +523,7 @@ func TestGenerateConfig(t *testing.T) {
 			actual, err := c.generateConfig()
 			require.NoError(err)
 
-			// If we got the arg handling write, verify output
-			golden := filepath.Join("testdata", tc.Name+".golden")
-			if *update {
-				ioutil.WriteFile(golden, actual, 0644)
-			}
-
-			expected, err := ioutil.ReadFile(golden)
-			require.NoError(err)
-			require.Equal(string(expected), string(actual))
+			golden.Assert(t, string(actual), tc.Name+".golden")
```

Comment:
`golden` provides the same flag (`-test.update-golden` instead of `-update`, otherwise the same idea).

The diff provided by this line uses the same underlying lib as testify (`difflib`), but has one additional feature. If the difference is only whitespace it will replace whitespace characters with visible character so that you can see the difference instead of staring at the screen trying to find the invisible difference.